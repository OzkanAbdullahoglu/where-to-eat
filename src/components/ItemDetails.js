import React from "react";
import { getCategoriesTitle } from "../utils/helper";

const ItemDetails = ({ cost, address, categories }) => (
  <div id="item-details">
    <span className="cards-result-cost">{cost}</span>
    <span className="cards-result-separator"> - </span>
    <span className="cards-result-details">
      {getCategoriesTitle(categories)}
    </span>
    <span className="cards-result-separator"> - </span>
    <span className="cards-result-location">{address}</span>
  </div>
);

export default ItemDetails;
