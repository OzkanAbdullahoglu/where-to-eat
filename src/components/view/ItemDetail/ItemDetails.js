/* eslint-disable linebreak-style */
import React from 'react';
import PropTypes from 'prop-types';
import { getCategoriesTitle } from '../../../utils/helper';

const ItemDetails = ({ cost, address, categories }) => (
  <div id="item-details">
    <span className="cards-result-cost">{cost}</span>
    <span className="cards-result-separator"> - </span>
    <span className="cards-result-details">
      {getCategoriesTitle(categories)}
    </span>
    <span className="cards-result-separator"> - </span>
    <span className="cards-result-location">{address}</span>
  </div>
);

ItemDetails.propTypes = {
  cost: PropTypes.string,
  address: PropTypes.string,
  categories: PropTypes.array,
};

export default ItemDetails;
