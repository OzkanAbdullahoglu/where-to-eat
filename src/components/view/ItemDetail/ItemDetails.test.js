/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ItemDetails from './ItemDetails';

describe('Item Details Component', () => {
  const props = {
    cost: '$$$$',
    address: 'test address',
    categories: [
      { title: 'Turkish', alias: 'turkish' },
    ],
  };
  const itemDetail = shallow(<ItemDetails {...props} />);
  it('renders Item Detail Component', () => {
    expect(toJson(itemDetail)).toMatchSnapshot();
  });
  it('render `cost` props', () => {
    const costSpan = itemDetail.find('.cards-result-cost').text();
    expect(costSpan).toEqual(props.cost);
  });
  it('render `address` props', () => {
    const addressSpan = itemDetail.find('.cards-result-location').text();
    expect(addressSpan).toEqual(props.address);
  });
  it('render `categories` props', () => {
    const categoriesSpan = itemDetail.find('.cards-result-details').text();
    expect(categoriesSpan).toEqual(props.categories[0].title);
  });
});

