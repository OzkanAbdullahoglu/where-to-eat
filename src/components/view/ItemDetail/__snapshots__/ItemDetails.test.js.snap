// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Details Component renders Item Detail Component 1`] = `
<div
  id="item-details"
>
  <span
    className="cards-result-cost"
  >
    $$$$
  </span>
  <span
    className="cards-result-separator"
  >
     - 
  </span>
  <span
    className="cards-result-details"
  >
    Turkish
  </span>
  <span
    className="cards-result-separator"
  >
     - 
  </span>
  <span
    className="cards-result-location"
  >
    test address
  </span>
</div>
`;
